asreview/datasets.py

```diff
@@ -0,0 +1,142 @@
+from pathlib import Path
+import pkg_resources
+
+from asreview.utils import pretty_format
+from asreview.utils import is_iterable
+
+
+class BaseDataSet():
+    name = "unknown"
+
+    def __init__(self, fp=None):
+        if fp is not None:
+            self.fp = fp
+            self.name = Path(fp).name
+
+    def get(self):
+        try:
+            return self.url
+        except AttributeError:
+            return self.fp
+
+    def to_dict(self):
+        mydict = {}
+        for attr in ["url", "fp", "authors", "topic", "link", "license"]:
+            try:
+                val = getattr(self, attr)
+                mydict[attr] = val
+            except AttributeError:
+                pass
+        return mydict
+
+    def __str__(self):
+        return pretty_format(self.to_dict())
+
+
+class PTSDDataSet(BaseDataSet):
+    name = "ptsd"
+    url = (
+        "https://raw.githubusercontent.com/asreview/systematic-review-datasets"
+        "/master/datasets/Van_de_Schoot_PTSD/output/PTSD_VandeSchoot_18.csv")
+    authors = ['Rens van de Schoot', 'Marit Sijbrandij', 'Sarah Depaoli',
+               'Sonja D. Winter', 'Miranda Olff', 'Nancy E. van Loey']
+    topic = "PTSD"
+    license = "CC-BY Attribution 4.0 International"
+    link = "https://osf.io/h5k2q/"
+    year = 2018
+
+
+class AceDataSet(BaseDataSet):
+    name = "ace"
+    url = (
+        "https://raw.githubusercontent.com/asreview/systematic-review-datasets"
+        "/master/datasets/Cohen_EBM/output/ACEInhibitors.csv")
+    link = ("https://dmice.ohsu.edu/cohenaa/"
+            "systematic-drug-class-review-data.html")
+    authors = ["A.M. Cohen", "W.R. Hersh", "K. Peterson", "Po-Yin Yen"]
+    year = 2006
+    license = None
+    topic = "ACEInhibitors"
+
+
+class HallDataSet(BaseDataSet):
+    name = "hall"
+    url = (
+        "https://raw.githubusercontent.com/asreview/systematic-review-datasets"
+        "/master/datasets/Four%20Software%20Engineer%20Data%20Sets/"
+        "Software%20Engineering%201%20Hall.csv"
+    )
+    link = "https://zenodo.org/record/1162952#.XIVBE_ZFyVR"
+    authors = ["Tracy Hall", "Sarah Beecham", "David Bowes", "David Gray",
+               "Steve Counsell"]
+    year = 2012
+    license = "CC-BY Attribution 4.0 International"
+    topic = "Software Fault Prediction"
+
+
+class ExampleDataSetGroup():
+    data_sets = [
```

Comment:
self? Not sure. Especially useful to make an isolated ExampleDataSetGroup?

Do we need an append method as well?

Reply:
I'm not quite sure what you mean. I added an append method, though I don't think it needs to be used really. Maybe with the update it makes more sense?

Reply:
No problem, if this works with an extension that adds datasets, I'm fine.